lib/serverengine/daemon_logger.rb

```diff
@@ -76,9 +85,51 @@ def add(severity, message = nil, progname = nil, &block)
       end
       progname ||= @progname
       self << format_message(SEVERITY_FORMATS_[severity+1], Time.now, progname, message)
+      poll_pending_inotify if defined?(INotify)
       true
     end
 
+    def poll_pending_inotify
+      return unless ServerEngine::linux?
+      return unless @inotify
+
+      if IO.select([@inotify.to_io], [], [], 0)
+        @inotify.process
+      end
+    end
+
+    def enable_watching_logdev(logdev)
+      return unless ServerEngine::linux?
+
+      target = nil
+      if logdev.respond_to?(:filename)
+        target = logdev.filename
+      elsif logdev.respond_to?(:path)
+        target = logdev.path
+      elsif logdev.is_a?(String)
+        target = logdev
+      else
+        # ignore StringIO for some test cases
+        return
+      end
+      if target
+        @inotify.close if @inotify
+        @inotify = INotify::Notifier.new
+        @inotify.watch(target, :move_self) do |event|
+          if logdev.respond_to?(:filename)
```

Comment:
Should it be `@logdev` instead of `logdev`? @kenhys

Reply:
Thanks, it should be `@logdev` in this context.

Reply:
Fixed.